Assets/MixedRealityToolkit-SDK/Features/SpatialAwareness/System/MixedRealitySpatialAwarenessSystem.cs

```diff
@@ -65,12 +66,17 @@ private GameObject CreateSecondGenerationParent(string name)
             return secondGeneration;
         }
 
-        // todo
+        private IMixedRealitySpatialAwarenessObserver spatialAwarenessObserver = null;
+
+        /// <summary>
+        /// The <see cref="IMixedRealitySpatialAwarenessObserver"/>, if any, that is active on the current platform.
+        /// </summary>
+        private IMixedRealitySpatialAwarenessObserver SpatialAwarenessObserver => spatialAwarenessObserver ?? (spatialAwarenessObserver = MixedRealityManager.Instance.GetManager<IMixedRealitySpatialAwarenessObserver>());
```

Comment:
Seems like this might be a `IMixedRealityComponent`? Although I think it's fine here at the moment. Thoughts?

Reply:
https://github.com/Microsoft/MixedRealityToolkit-Unity/pull/2848/files#diff-5f07c17e8951e44aaa4d837bbb50a0c7R12

Reply:
I'm wondering if we want to get it from the profile, or the manager (as discussed in the shiproom last week)

I don't think we had an in depth discussion about it. I could go either way.

Reply:
The observer will be a registered component. this is how I believe the input device managers work. Hence the design :)